torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -479,8 +562,26 @@ static void extra_kwargs(FunctionSignature& signature, PyObject* kwargs, ssize_t
   throw TypeError("invalid keyword arguments");
 }
 
-bool FunctionSignature::parse(PyObject* args, PyObject* kwargs, PyObject* dst[],
-                              bool raise_exception) {
+/*
+ * Like list.insert(), but for C arrays of PyObject*. Skips error checking.
+ */
+static void
+pyobject_array_insert(PyObject **array, int length, int index, PyObject *item)
```

Comment:
This seems like a code smell to me. This operation is `O(n)`, is it *really* necessary?

Reply:
Oh, this is for the subclass ordering logic